### PR TITLE
fuzzgen: Enable `bitcast` instruction

### PR DESCRIPTION
**Pull request by afonso360**
👋 Hey,

This PR enables the `bitcast` instruction in fuzzgen. This was previously fixed in #6379, and we can now start fuzzing it.

### CODE REVIEW

File: cranelift/fuzzgen/src/function_generator.rs [hunk 1]
```diff
@@ -454,15 +460,15 @@ fn valid_for_target(triple: &Triple, op: Opcode, args: &[Type], rets: &[Type]) -
...
+            let ret = rets[0];
 
             // Vector arguments must produce vector results, and scalar arguments must produce
             // scalar results.
             if arg.is_vector() != ret.is_vector() {
                 return false;
             }
 
-            if arg.is_vector() && arg.is_vector() {
+            if arg.is_vector() && ret.is_vector() {
```
Comment (afonso360) [inline cranelift/fuzzgen/src/function_generator.rs:471]: This is a little bit of a driveby fix, but these looked wrong. I ran fuzzgen afterwards and it didn't reveal any new panics.

**Review by alexcrichton — APPROVED** [review]
